### ISSUE TITLE
[Accessibility]: Provide more context for matching code lines in ref panel

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36582]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

When navigating the matches in a ref panel using a screen reader, it will read the line number and start reading the text. I guess after a bit, you can infer that it is talking about the line number followed by the content, but it seems a little odd.

### Expected behavior

Perhaps we should have the screen reader say "line number" before it announces the line number and "content" before it reads out the line content?

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
